### ISSUE TITLE
REST API ident query url encoding

### ISSUE THREAD

**Issue by p0ssi** [bootlin/elixir#217]
I'm trying to locate Linux driver files defining certain DT compatible strings using REST API ident query.

I have no local instance, so I'm running queries against elixir.bootlin.com, saying it's v2.1.

Query for identifier "gpio-key" yields expected results:
`curl "https://elixir.bootlin.com/api/ident/linux/gpio-key?version=latest&family=B" | python -m json.tool`

However, I get no results with (most typical) identifiers that include comma character, such as "linux,kvm":
`curl "https://elixir.bootlin.com/api/ident/linux/linux%2Ckvm?version=latest&family=B" | python -m json.tool`

Am I missing something?

**Comment by cxw42** [comment]
What's the raw server output (from curl, not piped through Python)?

**Comment by p0ssi** [comment]
Server responds 200, body looks like this:`{"definitions": [], "references": [], "documentations": []}`

**Comment by cxw42** [comment]
Hmmm... what if you quote the string in the query?  The code says the DB stores DT strings quoted.  https://github.com/bootlin/elixir/blob/master/api/api.py

**Comment by p0ssi** [comment]
No results with quotes either.

Noticed the same comment, the code adds quotes to the queried identifier, so it can be matched to database that also has quotes.

I get sane results for identifiers without comma character, so it makes me think the url decoding might not work correctly. Can't pinpoint a problem, though.

**Comment by p0ssi** [comment]
Just realized that parse.quote() in fact does not enclose the string in quotes as I misinterpreted the comment. It actually escapes the identifier with url encoding, replacing e.g. the comma character in with %2C.

I wonder whether that's intentional?

**Comment by carmeli-tamir** [comment]
Had some time to check this:

`curl -k "https://elixir.bootlin.com/api/ident/linux/gpio-key?version=latest&family=B"`

returns 

`definitions": [], "references": [{"path": "arch/arm/boot/dts/imx6ul-phytec-segin-peb-eval-01.dtsi", "line": "11", "type": null}], "documentations": [{"path": "Documentation/devicetree/bindings/vendor-prefixes.yaml", "line": "19", "type": null}]}`

https://elixir.bootlin.com/linux/latest/B/ident/gpio-key

The API matches the web page

Unless someone gets a different result, I think that the issue can be resolved

**Comment by p0ssi** [comment]
Yes, that works.

However, I could not figure out how to query identifiers containing comma character, such as "linux,kvm".

**Comment by carmeli-tamir** [comment]
Right, there's a bug that does url encodling twice - one in api.py and one in query.py. Submitted PR to remove the one in api.py, tested locally that the query "linux,kvm" work https://github.com/bootlin/elixir/pull/229

**Comment by fstachura** [comment]
The PR was merged and it that this is now fixed.
